src/main/java/org/jabref/logic/citationkeypattern/BracketedPattern.java

```diff
@@ -707,102 +707,97 @@ public static String lastAuthorForenameInitials(String authorField) {
     /**
      * Gets the last name of all authors/editors
      *
-     * @param authorField a <code>String</code>
+     * @param authorList an {@link AuthorList}
      * @return the sur name of all authors/editors
      */
-    public static String allAuthors(String authorField) {
-        // Quick hack to use NAuthors to avoid code duplication
-        return nAuthors(authorField, Integer.MAX_VALUE);
+    public static String allAuthors(AuthorList authorList) {
+        return joinAuthorsOnLastName(authorList, authorList.getNumberOfAuthors(), "", "");
     }
 
     /**
      * Returns the authors according to the BibTeX-alpha-Style
      *
-     * @param authorField string containing the value of the author field
-     * @return the initials of all authornames
+     * @param authorList an {@link AuthorList}
+     * @return the initials of all authors' names
      */
-    public static String authorsAlpha(String authorField) {
-        String authors = "";
+    public static String authorsAlpha(AuthorList authorList) {
+        StringBuilder authors = new StringBuilder();
 
-        String fixedAuthors = AuthorList.fixAuthorLastNameOnlyCommas(authorField, false);
+        String[] tokens = authorList.getAuthors().stream().map(Author::getLastOnly).toArray(String[]::new);
```

Comment:
Can probably be a lost

Reply:
Good point. I tried adding what is hopefully some readability improvements as well.